### ISSUE TITLE
JSpecify: nullable upper bound is lost when reading a generic field through ? super

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1854]
### Summary

In JSpecify mode, NullAway loses the nullable effective upper bound when a generic field is read through a `? super` receiver. Passing that field value to a non-null parameter produces no diagnostic.

This was found while investigating #1846. The method-return form in #1846 and this field-read form both have `@Nullable Comparable<Integer>` as the effective type of the value being read, but the field case is silently treated as non-null rather than crashing.

### Reproducer

Run with JSpecify mode and `OnlyNullMarked=true`:

```java
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

@NullMarked
class Test {
  static class Node<T extends @Nullable Comparable<Integer>> {
    T value;

    Node(T value) {
      this.value = value;
    }
  }

  static void takesObject(Object value) {}

  static void test(Node<? super Integer> node) {
    takesObject(node.value);
  }
}
```

Tested on commit `a1b3498f` with JDK 25.0.4.1, using the arguments from `JSpecifyJavacConfig.withJSpecifyModeArgs` plus `-XepOpt:NullAway:OnlyNullMarked=true`.

### Expected behavior

NullAway should report that `node.value` is nullable when it is passed to `takesObject`. Reading from `Node<? super Integer>` produces the wildcard's effective upper bound, which is `@Nullable Comparable<Integer>` from `T`'s declaration.

### Actual behavior

No diagnostic is reported.

### Investigation

Resolving `value` as a member of the receiver produces `? super Integer`. Its backing formal variable is `T`, whose upper bound is `@Nullable Comparable<Integer>`, and `GenericsUtils.effectiveWildcardUpperBound` correctly returns that annotated upper bound.

However, the field-read path does not use that effective substituted type when determining the expression's top-level nullness. `GenericsChecks.getTreeType` currently restores field annotations from the field symbol's declared type (`T`), and the main field/dataflow nullness path also does not propagate the nullable effective upper bound.

This is related to, but distinct from, #1846: that issue reaches annotation restoration with structurally mismatched wildcard and class types and crashes. This issue does not crash; it loses the nullable field-read nullness and yields a false negative.

Assisted-by: Codex (gpt-5)

**Comment by msridhar** [comment]
I think this is related to #1727 and may require proper modeling of `NO_CHANGE`
